codex-rs/core/src/protocol.rs

```diff
@@ -33,6 +33,8 @@ pub struct Submission {
 #[allow(clippy::large_enum_variant)]
 #[non_exhaustive]
 pub enum Op {
+    /// Erase all conversation history for the current session.
+    EraseConversationHistory,
```

Comment:
I think it would be cleaner if we introduced `Op::CompactConversationHistory` and then the TUI could send `AppEvent::CodexOp(Op::CompactConversationHistory)`, which ultimately should make it possible to eliminate that `rt.block_on()` code.

Again, this could be done in a separate PR and then this PR would wire up the `Op` in the TUI.